### PR TITLE
fix: Use annotation based column name

### PR DESCRIPTION
**Pull request by zhixuanjia**
## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Comment by zhixuanjia** [comment]
1. Fixed some unit tests. Now only 4 left. testListUrnsForSoftDeletedAspect x2 testListUrnsOffsetPagination x2
2. Moved to use annotation based column. Deleted unused code.
3. Minor fix here and there.

File: dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/utils/SQLStatementUtils.java [hunk 1]
```diff
@@ -60,6 +61,7 @@ public class SQLStatementUtils {
    *      ORDER BY i_testing_aspectfoo$value ASC)
    *    SELECT *, (SELECT count(urn) FROM _temp_results) AS _total_count FROM _temp_results
    */
+  // TODO: IMPORTANT - WITH expression only available since MySQL version 8. Rewrite this SQL to make it compatible with older version.
```
Comment (jsdonn) [inline dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/utils/SQLStatementUtils.java:64]: Did you already create a ticket for this?

File: dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalDAOTest.java [hunk 2]
```diff
@@ -431,11 +431,11 @@ public void testGetNonExisting() {
   }
 
   @Test
-  public void testGetCapsSensitivity() {
+  public void testGetCapsSensitivity() throws URISyntaxException {
     // This test will fail because Urn does not have an ENTITY_TYPE field which is required for EbeanDAOUtils::getEntityType
```
Comment (jsdonn) [inline dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalDAOTest.java:435]: can get rid of this comment

File: dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalDAOTest.java [hunk 3]
```diff
@@ -2972,41 +2961,39 @@ private void addIndex(Urn urn, String aspectName, String pathName, Object val, S
...
-    // these two columns have already been added in the setup script (metadata-schema-create-all.sql)
-    addedColumns.add("a_testing_aspectfoo");
-    addedColumns.add("a_testing_aspectbar");
-    String normalizedAspectName = SQLSchemaUtils.getNormalizedAspectName(aspectName); // e.g. com.linkedin.testing.AspectFoo -> testing_aspectfoo
-    String indexColumnName = SQLSchemaUtils.INDEX_PREFIX + normalizedAspectName; // e.g. testing_aspectfoo -> i_testing_aspectfoo
-    String aspectColumnName = SQLSchemaUtils.ASPECT_PREFIX + normalizedAspectName; // e.g. testing_aspectfoo -> a_testing_aspectfoo
-    String processedPathName = SQLSchemaUtils.processPath(pathName); // e.g. /path1/value1 -> $path1$value1
-    String fullIndexColumnName = indexColumnName + processedPathName; // e.g. i_testing_aspectfoo$path1$value1
+    String aspectColumnName = isUrn(aspectName) ? null : SQLSchemaUtils.getAspectColumnName(aspectName); // e.g. a_aspectfoo;
+    String fullIndexColumnName = SQLSchemaUtils.getGeneratedColumnName(aspectName, pathName); // e.g. i_testing_aspectfoo$path1$value1
```
Comment (jsdonn) [inline dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalDAOTest.java:2965]: will it still be `i_testing_aspectfoo$path1$value1`? or now it is just `i_aspectfoo$path1$value1`?

File: dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalDAOTest.java [hunk 4]
```diff
@@ -2972,41 +2961,39 @@ private void addIndex(Urn urn, String aspectName, String pathName, Object val, S
...
-          // since these columns have already been created in the metadata-schema-create-all.sql script as virtual columns,
-          // we need to drop them first so they can be regenerated in the way that the test(s) want it to be generated.
-          String sqlQuery = String.format("ALTER TABLE %s DROP COLUMN %s$value;", getTableName(urn), indexColumnName);
-          _server.execute(Ebean.createSqlUpdate(sqlQuery));
-        }
-        String sqlQuery = String.format("ALTER TABLE %s ADD COLUMN %s VARCHAR(255);",
-            getTableName(urn), fullIndexColumnName);
-        _server.execute(Ebean.createSqlUpdate(sqlQuery));
-        addedColumns.add(fullIndexColumnName);
+      String checkColumnExistance = "SELECT * FROM information_schema.COLUMNS WHERE TABLE_SCHEMA = 'gma_dev' AND"
```
Comment (jsdonn) [inline dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalDAOTest.java:2968]: can use DB_SCHEMA instead of gma_dev?

**Review by jsdonn — CHANGES_REQUESTED** [review]
Looks good overall, some minor things.

File: dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/utils/SQLStatementUtils.java (hunk 1 above)
Comment (zhixuanjia) [reply dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/utils/SQLStatementUtils.java:64]: Yes, created one https://jira01.corp.linkedin.com:8443/browse/META-16978

File: dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalDAOTest.java (hunk 2 above)
Comment (zhixuanjia) [reply dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalDAOTest.java:435]: updated.

File: dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalDAOTest.java (hunk 3 above)
Comment (zhixuanjia) [reply dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalDAOTest.java:2965]: `i_aspectfoo$path1$value1`. Updated.

File: dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalDAOTest.java (hunk 4 above)
Comment (zhixuanjia) [reply dao-impl/ebean-dao/src/test/java/com/linkedin/metadata/dao/EbeanLocalDAOTest.java:2968]: sure, updated.

**Review by jsdonn — APPROVED** [review]
LGTM.

**Pull request merged**